File: src/main/java/org/example/eventhub/controller/EventController.java
package org.example.eventhub.controller;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.example.eventhub.dto.event.EventCreateRequest;
import org.example.eventhub.dto.event.EventResponseLong;
import org.example.eventhub.dto.event.EventResponseShort;
import org.example.eventhub.dto.event.EventUpdateRequest;
import org.example.eventhub.service.EventService;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;


@RestController
@RequestMapping("/api/v1/events")
@RequiredArgsConstructor
public class EventController {

    private final EventService service;

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    public Page<EventResponseShort> getAllEvents(Pageable pageable) {
        return service.getAllEvents(pageable);
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public EventResponseLong createEvent(@Valid @RequestBody EventCreateRequest dto, @RequestParam Long organizerId) {
        return service.createEvent(dto, organizerId);
    }

    @GetMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public EventResponseLong getEventById(@PathVariable Long id) {
        return service.getEventById(id);
    }


    @PatchMapping("/{id}")
    @ResponseStatus(HttpStatus.OK)
    public EventResponseLong updateEvent(@PathVariable Long id, @Valid @RequestBody EventUpdateRequest dto) {
        return service.updateEvent(id, dto);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void deleteEvent(@PathVariable Long id) {
        service.deleteEvent(id);
    }
}

File: src/main/java/org/example/eventhub/controller/EventController.kt
package org.example.eventhub.controller

import jakarta.validation.Valid
import org.example.eventhub.dto.event.*
import org.example.eventhub.service.EventService
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.HttpStatus
import org.springframework.security.core.annotation.AuthenticationPrincipal
import org.springframework.security.core.userdetails.UserDetails
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/events")
class EventController(
    private val service: EventService
) {

    @GetMapping
    @ResponseStatus(HttpStatus.OK)
    fun getAllEvents(pageable: Pageable, eventFilter: EventFilter): Page<EventResponseShort> =
        service.getAllEvents(pageable, eventFilter)

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    fun createEvent(
        @Valid @RequestBody dto: EventCreateRequest,
        @AuthenticationPrincipal user: UserDetails
    ): EventResponseLong =
        service.createEvent(dto, user)

    @GetMapping("/{eventId}")
    @ResponseStatus(HttpStatus.OK)
    fun getEventById(@PathVariable eventId: Long): EventResponseLong =
        service.getEventById(eventId)

    @PatchMapping("/{eventId}")
    @ResponseStatus(HttpStatus.OK)
    fun updateEvent(
        @AuthenticationPrincipal user: UserDetails,
        @PathVariable eventId: Long,
        @Valid @RequestBody dto: EventUpdateRequest
    ): EventResponseLong =
        service.updateEvent(user, eventId, dto)

    @DeleteMapping("/{eventId}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun deleteEvent(@AuthenticationPrincipal user: UserDetails, @PathVariable eventId: Long) =
        service.deleteEvent(user, eventId)
}
